### PR TITLE
Decanonicalize labels emitted by {a,c,}query if possible

### PR DESCRIPTION
**Pull request by fmeum**
Uses the newly added `PackageIdentifier#getDisplayForm` to turn labels
in the output of query, aquery, and cquery into the most concise
representation that allows them to be resolved from the context of the
main repository.

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/cmdline/RepositoryMapping.java [hunk 1]
```diff
@@ -93,4 +95,17 @@ public RepositoryName get(String preMappingName) {
...
+    if (ownerRepo() == null) {
+      // This can only happen if Bzlmod is not enabled. In this case, assume no repo mappings are
+      // in use, so that canonical and  local names can be used interchangeably.
+      return Optional.of(postMappingName.getName());
+    }
+    Preconditions.checkArgument(ownerRepo().isMain());
+    return repositoryMapping().entrySet().stream()
+        .filter(e -> e.getValue().equals(postMappingName))
+        .map(Entry::getKey)
+        .findFirst();
```
Comment (fmeum) [inline src/main/java/com/google/devtools/build/lib/cmdline/RepositoryMapping.java:109]: This is `O(num mappings)`, so a proper implementation would instead store the mapping in an `ImmutableSetMultimap`.

**Comment by fmeum** [comment]
Stacked on #16482 and thus in draft state, but otherwise ready for review.

**Comment by fmeum** [comment]
I pushed a commit that updates more formatters. 

This PR doesn't cover the `BUILD`-like output modes, which would have to decanonicalize labels with respect to the current rather than the main repository. That would require more extensive changes though.

**Comment by fmeum** [comment]
@Wyverald I rebased onto master, this is ready for review now.

File: src/main/java/com/google/devtools/build/lib/cmdline/Label.java [hunk 2]
```diff
@@ -416,9 +416,21 @@ public String getCanonicalForm() {
...
+   * Returns a label string that is suitable for display, i.e., it resolves to this label when
+   * parsed in the context of the main repository and has a repository part that is as simple as
+   * possible.
+   *
+   * @param mainOrOtherRepoMapping the {@link RepositoryMapping} of the main repository or, if that
+   *                               is not available to the caller, the {@link RepositoryMapping} of
+   *                               the current package. The former will yield more readable output.
+   * @return analogous to {@link PackageIdentifier#getDisplayForm(RepositoryMapping)}
+   */
+  public String getDisplayForm(RepositoryMapping mainOrOtherRepoMapping) {
```
Comment (Wyverald) [inline src/main/java/com/google/devtools/build/lib/cmdline/Label.java:432]: this should also be just `mainRepoMapping`, right?

File: src/main/java/com/google/devtools/build/lib/query2/engine/QueryEnvironment.java [hunk 3]
```diff
@@ -536,6 +537,10 @@ ThreadSafeMutableSet<T> getBuildFiles(
    */
   TargetAccessor<T> getAccessor();
 
+  default RepositoryMapping getMainRepoMapping() {
```
Comment (Wyverald) [inline src/main/java/com/google/devtools/build/lib/query2/engine/QueryEnvironment.java:540]: does this need to have a default?

File: src/main/java/com/google/devtools/build/lib/cmdline/Label.java (hunk 2 above)
Comment (fmeum) [reply src/main/java/com/google/devtools/build/lib/cmdline/Label.java]: Fixed, thanks!

**Review by Wyverald — APPROVED** [review]

**Comment by fmeum** [comment]
@bazel-io flag

**Comment by kshyanashree** [comment]
Hello @fmeum! As I could see we are good to merge the PR, does it require any other review? Thanks!

**Comment by fmeum** [comment]
@kshyanashree Yes, this is good to merge!

**Comment by meteorcloudy** [comment]
@bazel-io fork 6.0.0

**Pull request closed without merging**